providers/httpnet/api.go

```diff
@@ -0,0 +1,260 @@
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/StackExchange/dnscontrol/v3/pkg/diff"
+	"golang.org/x/net/idna"
+)
+
+const endpoint = "https://partner.routing.net/api/%s/v1/json/%s"
+
+type api struct {
+	authToken      string
+	ownerAccountID string
+}
+
+func (api *api) getDomainConfig(domain string) (*domainConfig, error) {
+	zc, err := api.getZoneConfig(domain)
+	if err != nil {
+		return nil, fmt.Errorf("error getting zone config: %v", err)
+	}
+
+	params := request{
+		Filter: filter{
+			Field: "domainName",
+			Value: zc.Name,
+		},
+	}
+
+	resp, err := api.get("domain", "domainsFind", params)
+	if err != nil {
+		return nil, fmt.Errorf("error getting domain info: %v", err)
+	}
+
+	domainConf := []*domainConfig{}
+	if err := json.Unmarshal(resp.Data, &domainConf); err != nil {
+		return nil, fmt.Errorf("error parsing response: %v", err)
+	}
+
+	if len(domainConf) == 0 {
+		return nil, fmt.Errorf("could not get domain config: %s", domain)
+	}
+
+	return domainConf[0], nil
+}
+
+func (api *api) createZone(domain string) error {
+	t, err := idna.ToASCII(domain)
+	if err != nil {
+		return err
+	}
+
+	records := []*record{}
+	for _, ns := range defaultNameservers {
+		records = append(records, &record{
+			Name:    domain,
+			Type:    "NS",
+			Content: ns,
+			TTL:     86400,
+		})
+	}
+
+	params := request{
+		ZoneConfig: &zoneConfig{
+			Name: t,
+			Type: "NATIVE",
+		},
+		Records: records,
+	}
+
+	_, err = api.get("dns", "zoneCreate", params)
+	if err != nil {
+		return fmt.Errorf("error creating zone: %v", err)
+	}
+	return nil
+}
+
+func (api *api) getNameservers(domain string) ([]string, error) {
+	t, err := idna.ToASCII(domain)
+	if err != nil {
+		return nil, err
+	}
+
+	domainConf, err := api.getDomainConfig(t)
+	if err != nil {
+		return nil, fmt.Errorf("error getting domain config: %v", err)
+	}
+
+	nss := []string{}
+	for _, ns := range domainConf.Nameservers {
+		// Currently does not support glued IP addresses
+		if len(ns.IPs) > 0 {
+			return nil, fmt.Errorf("domain %s has glued IP addresses which are not supported", domain)
+		}
+
+		nss = append(nss, ns.Name)
+	}
+
+	return nss, nil
+}
+
+func (api *api) updateNameservers(nss []string, domain string) func() error {
+	return func() error {
+		domainConf, err := api.getDomainConfig(domain)
+		if err != nil {
+			return err
+		}
+
+		nameservers := []nameserver{}
+		for _, ns := range nss {
+			nameservers = append(nameservers, nameserver{Name: ns})
+		}
+
+		domainConf.Nameservers = nameservers
+
+		params := request{
+			Domain: domainConf,
+		}
+
+		if _, err := api.get("domain", "domainUpdate", params); err != nil {
+			return err
+		}
+		return nil
+	}
+}
+
+func (api *api) getRecords(domain string) ([]*record, error) {
+	zc, err := api.getZoneConfig(domain)
+	if err != nil {
+		return nil, err
+	}
+
+	params := request{
+		Filter: filter{
+			Field: "ZoneConfigId",
+			Value: zc.ID,
+		},
+		Limit: 1000,
+	}
+
+	// TODO: Support more than 1000 records
```

Comment:
Please be sure to enable "pager1201" in `integrationTest/integration_test.go` when this is implemented.  (For testing purposes, feel free to reduce the limit to something smaller and enable "pager101" or "pager601")

Reply:
Implemented and passing.